app/models/incomeSourceDetails/TaxYear.scala

```diff
@@ -49,5 +49,15 @@ object TaxYear {
       case _ => None
     }
   }
+
+  def isValidTaxYear(year: Option[TaxYear], latencyDetails: Option[LatencyDetails]): Boolean = {
```

Comment:
function should look like:
`isValidTaxYear(year: TaxYear, latencyDetails: LatencyDetails): Boolean`

where these are Optional you can use flatMap or for-comprehension to use main function